clang/lib/AST/StmtProfile.cpp

```diff
@@ -2257,6 +2257,11 @@ void StmtProfiler::VisitSubstNonTypeTemplateParmExpr(
     const SubstNonTypeTemplateParmExpr *E) {
   // Profile exactly as the replacement expression.
   Visit(E->getReplacement());
+  if (auto *NTTP =
+          dyn_cast_if_present<NonTypeTemplateParmDecl>(E->getParameter());
```

Comment:
In `SubstNonTypeTemplateParmExpr::getParameter()` we returned a `NonTypeTemplateParmDecl` after a cast; so an `if_present` is unnecessary, at least `E->getParameter()` was presumed to return always non-null.